spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -822,28 +866,14 @@ function timeoutChannelUpgrade(
     // get underlying connection for proof verification
     connection = getConnection(currentChannel.connectionIdentifier)
 
-    // counterparty channel must be proved to still be in OPEN state or INITUPGRADE state (crossing hellos)
-    abortTransactionUnless(counterpartyChannel.State === OPEN || counterpartyChannel.State == INITUPGRADE)
+    // counterparty channel must be proved to not have completed flushing after timeout has passed
+    abortTransactionUnless(counterpartyChannel.state !== OPEN || counterpartyChannel.state == FLUSHCOMPLETE)
```

Comment:
I think this check needs to be modified. This should only state that counterpartyChannel must be in any state except FLUSHCOMPLETE?

Reply:
It could have completed handshake and went to OPEN. added more logic to check if this indeed did happen

Reply:
see latest comment on this https://github.com/cosmos/ibc/pull/999#discussion_r1276033187